### PR TITLE
docs: add icon contribution guidelines

### LINKED ISSUE #1907: [DOCS] Contributing icons docs
### Description

Add a file to the docs/contributing folder called contributing-icons.md

In the file give full instructions for how to fork and clone the project, add the icons to the following 3 files:
icons.md
GetIcons.js
links.json

Then add instuctions for how to commit and add the correct files and make a pull request

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by virtual-designer**
## Fixes Issue
Closes #1907 

## Changes proposed

* Added icon contribution guidelines

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/1911"></a>

### CODE REVIEW

**Review by Shubhdeep12 — APPROVED** [review]

File: docs/contributing/contributing-icons.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+```
+git clone https://github.com/<your-github-username>/LinkFree
+```
+
+Replace `<your-github-username>` with your GitHub username.
+
+## How to add icons
+
+To add icons, you need to modify three files: [GetIcons.js](https://github.com/EddieHubCommunity/LinkFree/blob/main/src/Components/Icons/GetIcons.js), [icons.md](https://github.com/EddieHubCommunity/LinkFree/blob/main/icons.md), [links.json](https://github.com/EddieHubCommunity/LinkFree/blob/main/src/config/links.json).
```
Comment (Panquesito7) [inline docs/contributing/contributing-icons.md:19]:
```suggestion
To add a new icon, you need to modify the following files: [`GetIcons.js`](https://github.com/EddieHubCommunity/LinkFree/blob/main/src/Components/Icons/GetIcons.js), [`icons.md`](https://github.com/EddieHubCommunity/LinkFree/blob/main/icons.md), and [`links.json`](https://github.com/EddieHubCommunity/LinkFree/blob/main/src/config/links.json).
```

File: docs/contributing/contributing-icons.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## How to add icons
+
+To add a new icon, you need to modify the following files: [`GetIcons.js`](https://github.com/EddieHubCommunity/LinkFree/blob/main/src/Components/Icons/GetIcons.js), [`icons.md`](https://github.com/EddieHubCommunity/LinkFree/blob/main/icons.md), and [`links.json`](https://github.com/EddieHubCommunity/LinkFree/blob/main/src/config/links.json).
+
+The `GetIcons.js` file contains the main code that shows the icons on the page. It has a `GetIcons` component which takes a `iconName` prop and returns an icon according to that. You need to add another case inside the switch statement in the component for the new icon.
+
+The `icons.md` file is the index of all available icons. If you add a new icon, be sure to update this index so that people can see what's the code of the icon and how does it look like.
+
+The `links.json` file has `validIcons` object which specifies the color codes for the icons. Be sure to add the correct color for the new icon!
```
Comment (Panquesito7) [inline docs/contributing/contributing-icons.md:25]: Hmm, maybe we could add an example of adding a new icon? For example: adding the Spotify icon, or any other icon (it can be in the list, too, just needs to be an example). What do you think? 🙂

Comment (virtual-designer) [reply docs/contributing/contributing-icons.md:25]: I've added that!

**Review by E-fais — COMMENTED** [review]
LGTM ;)

**Review by Panquesito7 — COMMENTED** [review]
Nice work! 🚀

File: docs/contributing/contributing-icons.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+|   Android   |   `android`   | <img src="https://user-images.githubusercontent.com/65664185/138502465-89cfadf2-6b54-4f3d-ac44-ceacdd4824ba.png" width=65% height=30%>                       |     Apple      |     `apple`      | <img src="https://user-images.githubusercontent.com/65664185/138502540-8e9b80bf-deae-4566-a41a-c63623e83c21.png" width=100% height=30%> |
+|    Book     |    `book`     | <img src="https://user-images.githubusercontent.com/76985777/145391108-f8c08f8e-679f-45a3-ad58-83ef60aa28fe.png" width=65% height=30%>                       |   Codeforces   |   `codeforces`   | <img src="https://user-images.githubusercontent.com/91655303/148160942-870fdbb4-a57c-4861-afaa-241835390645.png" width=100% height=30%> |
+
+<!-- ... -->
+
+|   YouTube   |   `youtube`   | <img src="https://user-images.githubusercontent.com/65664185/138503305-ff60cf54-6b0b-4e18-9446-b4f6869b9511.png" width=65% height=30%>                       |    Spotify     |    `spotify`     | <img src="<icon_preview_url_here>" width="100%" height="30%">                                                                           |
+```
+
+And that's it!
```
[the remark this answers is not in the document]
Comment (virtual-designer) [reply docs/contributing/contributing-icons.md]: Looks good! Changes committed!

File: docs/contributing/contributing-icons.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+```
+git clone https://github.com/<your-github-username>/LinkFree
+```
+
+Replace `<your-github-username>` with your GitHub username.
+
+## How to add icons
+
+To add a new icon, you need to modify the following files: [`GetIcons.js`](https://github.com/EddieHubCommunity/LinkFree/blob/main/src/Components/Icons/GetIcons.js), [`icons.md`](https://github.com/EddieHubCommunity/LinkFree/blob/main/icons.md), and [`links.json`](https://github.com/EddieHubCommunity/LinkFree/blob/main/src/config/links.json).
+
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:20]: Add a heading here: What to add in GetIcons.js

File: docs/contributing/contributing-icons.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+The `GetIcons.js` file contains the main code that shows the icons on the page. It has a `GetIcons` component which takes a `iconName` prop and returns an icon according to that. You need to add another case inside the switch statement in the component for the new icon.
+
+The `icons.md` file is the index of all available icons. If you add a new icon, be sure to update this index so that people can see what's the code of the icon and how does it look like.
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:23]: Add heading: What to add in icons.md

File: docs/contributing/contributing-icons.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:24]: Add heading: What to add in links.js

File: docs/contributing/contributing-icons.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+The `links.json` file has `validIcons` object which specifies the color codes for the icons. Be sure to add the correct color for the new icon!
+
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:26]: Add heading: Example of adding an icon

File: docs/contributing/contributing-icons.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+For example, if you want to add "Spotify" icon, you will need to add the following import statements at the top of the `GetIcons.js` file:
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:27]:
```suggestion
1. To add the "Spotify" icon, add the following import statements at the top of the `GetIcons.js` file:
```

File: docs/contributing/contributing-icons.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  FaAndroid,
+  FaApple,
+  FaArrowLeft,
+  FaBook,
+  // ...
+  FaSpotify
+} from 'react-icons/fa'
+```
+
+The react icons library provides you icons from some other libraries like Font Awesome, Heroicons, Material Icons, etc. If you want to use them, make sure to use the correct imports! More information about react icons library can be found [here](https://react-icons.github.io/react-icons/).
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:40]:
```suggestion
The react icons library provides access to icons from several libraries like Font Awesome, Heroicons, Material Icons, etc. To use them, make sure to use the correct imports! More information about react icons library can be found [here](https://react-icons.github.io/react-icons/).
```

File: docs/contributing/contributing-icons.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+After adding imports, you need to add a new case to the switch statement inside of the `GetIcons` component:
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:42]:
```suggestion
2. Add a new case to the switch statement in alphabetical order inside of the `GetIcons` component:
```

File: docs/contributing/contributing-icons.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    // spotify icon
+    // if `iconName` is "spotify", then it will return this icon
+    case 'spotify':
+      return <FaSpotify {...restProps} />
+    // ...
+  }
+}
+```
+
+After that open `links.json` file and add a new property to the `validIcons` object. The property name must be same as the icon name (in this case, `spotify`).
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:60]:
```suggestion
3. Open `links.json` file and add a new property to the `validIcons` object. The property name must be same as the icon name (in this case, `spotify`).
```

File: docs/contributing/contributing-icons.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+```json
+{
+   "validIcons": {
+      "android": "#3DDC84",
+      "spotify": "#1DB954"
+   }
+}
+```
+
+Then open `icons.md` file and add the new icon name, usage and preview image to the icon table:
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:71]:
```suggestion
4. Open `icons.md` file and add the new icon name, usage and preview image to the icon table in alphabetical order:
```

File: docs/contributing/contributing-icons.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+|   Android   |   `android`   | <img src="https://user-images.githubusercontent.com/65664185/138502465-89cfadf2-6b54-4f3d-ac44-ceacdd4824ba.png" width=65% height=30%>                       |     Apple      |     `apple`      | <img src="https://user-images.githubusercontent.com/65664185/138502540-8e9b80bf-deae-4566-a41a-c63623e83c21.png" width=100% height=30%> |
+|    Book     |    `book`     | <img src="https://user-images.githubusercontent.com/76985777/145391108-f8c08f8e-679f-45a3-ad58-83ef60aa28fe.png" width=65% height=30%>                       |   Codeforces   |   `codeforces`   | <img src="https://user-images.githubusercontent.com/91655303/148160942-870fdbb4-a57c-4861-afaa-241835390645.png" width=100% height=30%> |
+
+<!-- ... -->
+
+|   YouTube   |   `youtube`   | <img src="https://user-images.githubusercontent.com/65664185/138503305-ff60cf54-6b0b-4e18-9446-b4f6869b9511.png" width=65% height=30%>                       |    Spotify     |    `spotify`     | <img src="<icon_preview_url_here>" width="100%" height="30%">                                                                           |
+```
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:86]: Put this in alphabetical order

File: docs/contributing/contributing-icons.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+And that's it! You can now commit, push, and [submit a PR](htttps://github.com/EddieHubCommunity/LinkFree/compare), and the maintainers will check your PR. 🙂
+
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:89]: Remove this section, people won't continue to read the rest of the instructions.

File: docs/contributing/contributing-icons.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+## Committing and pushing the changes
+
+After you've finished making changes, you need to commit and push those first. 
+If you've cloned the forked repo locally and working on it, run these commands:
+
+```
+git add <files_you_have_changed>
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:96]: specify the exact git add command here so that noone is left guessing what to write for the files_you_have_changed

File: docs/contributing/contributing-icons.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+git commit -m "<commit_message>"
+git push origin main
+```
+
+This will push changes to the forked repo.
+
+Otherwise if you're working on GitHub website, then you can just save the changes and then it will commit changes for you. 
+
+After all of these, you can go to your forked repo, and click on "Contribute" > "Create Pull Request". This will take you to the Pull Request creation page. Most of the things are automatically adjusted there so most of the times you don't need to change anything there. Then click "Create Pull Request".
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:105]: Can this be made more specific. If I were a new contributor I wouldn't know what to do with this information. This is still quite a few steps that need to be explained so folks know which buttons to press, which branches to merge into which etc.

**Review by EmmaDawsonDev — CHANGES_REQUESTED** [review]
This is a great start! I've made a few suggestions to increase the clarity of the instructions as it's often beginners who try to contribute icons.

File: docs/contributing/contributing-icons.md (hunk 7 above)
Comment (EmmaDawsonDev) [reply docs/contributing/contributing-icons.md:23]: Just this heading to add then I think this is ready to merge.

**Comment by virtual-designer** [comment]
Added missing heading.

File: docs/contributing/contributing-icons.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
+# Contributing to the Icons
+
+We're glad to know you've decided to contribute!
+You can get started by forking the repo or cloning it locally.
+
+You can fork the repo by [clicking here](https://github.com/EddieHubCommunity/LinkFree/fork). Then create a branch for the issue that you're working on. After that, commit your changes. For more detailed information, see the [Contribution Guidelines](https://github.com/virtual-designer/LinkFree/blob/patch-4/docs/contributing/CONTRIBUTING.md#-how-to-contribute).
```
Comment (kumarsonsoff3) [inline docs/contributing/contributing-icons.md:6]: I recommend using the main link for the `contributing.md` instead of the one of your fork! 

```suggestion
You can fork the repo by [clicking here](https://github.com/EddieHubCommunity/LinkFree/fork). Then create a branch for the issue that you're working on. After that, commit your changes. For more detailed information, see the [Contribution Guidelines](https://github.com/EddieHubCommunity/LinkFree/blob/main/docs/contributing/CONTRIBUTING.md).
```

File: docs/contributing/contributing-icons.md [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+Otherwise if you're working on GitHub website, first make sure that you're on the correct branch. If you don't have one, you can create it while you commit the changes.
+
+After that, do the following:
+
+1. Scroll down to the "Commit changes" section at the bottom of the page.
+2. Check if you have the correct branch selected. Otherwise you can click on "Create a new branch and start a pull request" option. This will create a new branch with the changes when you make the commit.
+3. Adjust the commit message.
+4. Click the "Commit Changes" button at the bottom of the page and then it will commit the changes for you.
+
+After all of these, do the following:
```
Comment (kumarsonsoff3) [inline docs/contributing/contributing-icons.md:113]: I recommend highlighting this line for visibility reasons, what do you think!?
```suggestion
**After all of these, do the following:**
```

File: docs/contributing/contributing-icons.md [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+git add src/config/links.json src/Components/Icons/GetIcons.js icons.md
+git commit -m "<commit_message>"
+git push origin main
+```
+
+This will push changes to the forked repo.
+
+Otherwise if you're working on GitHub website, first make sure that you're on the correct branch. If you don't have one, you can create it while you commit the changes.
+
+After that, do the following:
```
Comment (kumarsonsoff3) [inline docs/contributing/contributing-icons.md:106]: I recommend highlighting this line for visibility reasons, what do you think!?

```suggestion
**After that, do the following:**
```

**Review by kumarsonsoff3 — CHANGES_REQUESTED** [review]
These are the last ones, Hope it will be good to go after these! 😉

File: docs/contributing/contributing-icons.md (hunk 19 above)
Comment (virtual-designer) [reply docs/contributing/contributing-icons.md:106]: Um to be honest I think it's okay without highlighting.

Comment (kumarsonsoff3) [reply docs/contributing/contributing-icons.md:106]: Hmm, true, but in the markdown format it looks too wordy. 
Also, you know people are hard at reading docs! 😄 So I think of highlighting these to enhance the visibility.
image

File: docs/contributing/contributing-icons.md [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+
+## Committing and pushing the changes
+
+After you've finished making changes, you need to commit and push those first. 
+If you've cloned the forked repo locally and working on it, run these commands:
+
+```
+git add src/config/links.json src/Components/Icons/GetIcons.js icons.md
+git commit -m "<commit_message>"
+git push origin main
```
Comment (EmmaDawsonDev) [inline docs/contributing/contributing-icons.md:99]: One small change - I really hope they’re pushing to a branch and not to main 

```suggestion
git push origin <branch name>
```

**Comment by EmmaDawsonDev** [comment]
I think let’s just get this merged and then create new issues to fix any remaining problems.
